zerver/tornado/event_queue.py

```diff
@@ -955,10 +955,11 @@ def get_client_payload(apply_markdown: bool, client_gravatar: bool) -> Dict[str,
             and "wildcard_mentioned" in flags
             and "read" not in flags
         )
+        sender_is_muted = user_data.get("sender_is_muted", False)
 
         # We first check if a message is potentially mentionable,
         # since receiver_is_off_zulip is somewhat expensive.
-        if (
+        if (not sender_is_muted) and (
```

Comment:
I think this would read more clearly as a `if sender_is_muted: continue` before this block -- it's best to avoid making individual if statements have really complicated logic from a readability standpoint.

Reply:
Makes sense. Updated to use that pattern.